src/extension.ts

```diff
@@ -64,20 +64,32 @@ export function activate(context: vscode.ExtensionContext) {
             await node.restart().then(() => outputChannel.appendLine(`App "${node.site.name}" has been restarted.`), err => outputChannel.appendLine(err));
         }
     });
+    initAsyncCommand(context, 'appService.Delete', async (node: AppServiceNode) => {
+        if (node) {
+            outputChannel.appendLine(`Deleting App "${node.site.name}"...`);
+            let result = await node.delete(azureAccount);
+            if (result) {
+                outputChannel.appendLine(`App "${node.site.name}" has been deleted.`), err => outputChannel.appendLine(err);
```

Comment:
I'm guessing you copied from the functions above, but those aren't implemented the best way. Start, stop, and restart actually return a nested promise. Instead of
```
async restart(): Promise<void> {
        await this.stop();
        return this.start();
    }
```
It should be:
```
async restart(): Promise<void> {
        await this.stop();
        await this.start();
    }
```

And then this file could be changed to:
```
await node.stop();
outputChannel.appendLine(`App "${node.site.name}" has been stopped.`)
```

If you want to fix those while you're here that'd be great :) Otherwise I can do it later